crates/sncast/src/response/declare.rs

```diff
@@ -129,3 +135,103 @@ impl OutputLink for DeclareTransactionResponse {
         )
     }
 }
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct DeployCommandMessage {
+    accounts_file: Option<String>,
+    account: String,
+    class_hash: PaddedFelt,
+    arguments: Option<String>,
+    network_flag: String,
+}
+
+impl DeployCommandMessage {
+    pub fn new(
+        abi: &[AbiEntry],
+        response: &DeclareTransactionResponse,
+        account: &str,
+        accounts_file: &Utf8PathBuf,
+        rpc_url: Option<&str>,
+        network: Option<&Network>,
+    ) -> Result<Self, Error> {
+        let arguments = abi.iter().find_map(|entry| {
+            if let AbiEntry::Constructor(constructor) = entry {
+                let args = generate_constructor_placeholder_args(constructor.clone());
+                (!args.is_empty()).then_some(args)
+            } else {
+                None
+            }
+        });
+        let network_flag = generate_network_flag(rpc_url, network);
+        let accounts_file_str = accounts_file.to_string();
+        let accounts_file = (!accounts_file_str
+            .contains("starknet_accounts/starknet_open_zeppelin_accounts.json"))
+        .then_some(accounts_file_str);
+
+        Ok(Self {
+            account: account.to_string(),
+            accounts_file,
+            class_hash: response.class_hash,
+            arguments,
+            network_flag,
+        })
+    }
+}
```

Comment:
Nit, but some of these are using a helper functions and some not, I think it would be a bit more readable if logic for generating `arguments` and `accounts_file` was moved to a helper as well.